### PR TITLE
Add Forumoji addon

### LINKED ISSUE #4097: Emodjiango (might turn Scratchmoji soon)
### Additional context

I have a Scratch forum topic where we create Scratch-forum-style emojis, and we're trying to make it of all the Unicode emojis.

Since emojis are not supported on the forums, it'd be neat if we could add an addon that replaces the normal emojis with the Scratch ones.

I'm working on a JSON file with most of the codepoints of the current one, and the repository with all the emojis is here: https://github.com/lopste/emodjiango (most recent emojis)

And the scratch forum topic is here: https://scratch.mit.edu/discuss/topic/557083/

### PR DESCRIPTION
**Pull request by mybearworld**
Resolves #4097

### Changes

Add an addon called "Forumoji emoji when posting on the forums" (internally `forumoji`) that replaces emojis with their Forumoji variants.

The forumoji addon.

### Reason for changes

Two reasons:
1. It's more convenient than having to go to the Forumoji web page and having to search for the emoji you want.
2. It avoids accidentally putting an emoji into the post to mess up the page's markup.

### Tests

Tested on Microsoft Edge 114.0.1807.6 (Official build) dev (64-bit).

### CODE REVIEW

**Comment by mybearworld** [comment]
I am currently just cloning the Post button, which means the text of the Post button will be in the user language of Scratch automatically. Should I use a localization string instead? It seems unnecessary, but I'm not sure.

**Comment by WorldLanguages** [comment]
> I am currently just cloning the Post button, which means the text of the Post button will be in the user language of Scratch automatically.

That is good practice. No need to create a localization string.

**Comment by mybearworld** [comment]
what the heck

**Comment by mybearworld** [comment]
NO

**Comment by DNin01** [comment]
If you pull upstream on all of your PR branches it should stop formatting unexpectedly.

**Comment by lankybox02** [comment]
image
Ehh... could the name & option title be a bit shorter?

"Forumoji emoji when posting on the forums" => "Unicode to Forumoji"
"Action if emoji are not in forumoji" => imo, this shouldn't be an option and should just ask by default
But also an emoji picker would be nice, that was in my original issue

**Comment by mybearworld** [comment]
> Prettier got a major update recently. If you pull upstream on all of your PR branches it should stop formatting unexpectedly.

Thanks, I've done that.

**Comment by mybearworld** [comment]
> "Forumoji emoji when posting on the forums" => "Unicode to Forumoji"

Maybe "Emoji to Forumoji"?



> "Action if emoji are not in forumoji" => imo, this shouldn't be an option and should just ask by default

That does make sense.



> But also an emoji picker would be nice, that was in my original issue

Couldn't you use your native emoji picker?

**Comment by Secret-chest** [comment]
> Couldn't you use your native emoji picker?

linux

**Comment by mybearworld** [comment]
> > Couldn't you use your native emoji picker?
> 
> linux

Ah, okay. I'll make a picker.

**Comment by Samq64** [comment]
> > Couldn't you use your native emoji picker?
> 
> linux

Don't Gnome and Plasma have  built-in colour pickers? And if you're using something else I'm sure there's a bunch of third-party ones to choose from.

**Comment by mybearworld** [comment]
> Ah, okay. I'll make a picker.

I have one implemented, *but* the images load extremely slowly, and this holds off the load of every page by over ten seconds. This isn't ideal.

**Comment by Secret-chest** [comment]
> Don't Gnome and Plasma have built-in colour pickers? And if you're using something else I'm sure there's a bunch of third-party ones to choose from.

Find me one for Cinnamon

**Comment by Samq64** [comment]
> Find me one for Cinnamon

There might be one built in. Have you tried pressing Super+.?

**Comment by Secret-chest** [comment]
> Find me one for Cinnamon
> 
> There might already be one. Have you tried pressing Super + .?

No, there isn't one.

**Comment by lankybox02** [comment]
> > Ah, okay. I'll make a picker.
> 
> I have one implemented, _but_ the images load extremely slowly, and this holds off the load of every page by over ten seconds. This isn't ideal.

they should be paginated?

**Comment by mybearworld** [comment]
Couldn't you just go to the forumoji site itself for an emoji picker? I'm not sure if it makes sense for this addon to remake that.

File: addons-l10n/en/forumoji.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
```
[the remark this answers is not in the document]
Comment (mybearworld) [reply addons-l10n/en/forumoji.json:1]: Pretty sure "emoji" stays the same in the plural

**Pull request closed without merging**